### PR TITLE
Improve docs for new apis

### PR DESCRIPTION
**Pull request by aboedo**
improved docs for `beginRefundRequest` and `showManageSubscriptions`.

### CODE REVIEW

File: Purchases/Purchasing/Purchases.swift [hunk 1]
```diff
@@ -1260,25 +1260,41 @@ public extension Purchases {
...
+     * the App Store's subscription management section will be opened.
+     *
+     * The `completion` block will be called when the modal is opened, not when it's actually closed.
+     * This is because of an undocumented change in StoreKit's behavior between iOS 15.0 and 15.2,
+     * where 15.0 would return when the modal was closed,
+     * and 15.2 returns when the modal is opened.
      */
     @available(watchOS, unavailable)
     @available(tvOS, unavailable)
+    @available(iOS 13.0, macOS 10.15, *)
```
Comment (aboedo) [inline Purchases/Purchasing/Purchases.swift:1278]: this was causing the docs to mark the method as deprecated

File: Purchases/Purchasing/Purchases.swift [hunk 2]
```diff
@@ -1260,25 +1260,41 @@ public extension Purchases {
...
      * Use this function to open the manage subscriptions modal.
-     * If the manage subscriptions modal can't be opened, the managementURL in the customerInfo will be opened.
-     * If managementURL is not available, the App Store's subscription management section will be opened.
      *
-     * - Parameter completion: A completion block that is called when the modal is closed.
-     * If it was not successful, there will be an `Error`.
+     * - Parameter completion: A completion block that is called when the modal is opened.
+     * - throws: an Error will be thrown if the current window scene couldn't be opened,
+     * or the ``CustomerInfo/managementURL`` couldn't be obtained.
+
```
Comment (taquitos) [inline Purchases/Purchasing/Purchases.swift:1289]: missing `*` for this new line

**Review by taquitos — APPROVED** [review]
Tiny update, but then 🚢

File: Purchases/Purchasing/Purchases.swift (hunk 1 above)
Comment (NachoSoto) [reply Purchases/Purchasing/Purchases.swift:1278]: Oooh good point.

**Review by NachoSoto — APPROVED** [review]

File: Purchases/Purchasing/Purchases.swift (hunk 2 above)
Comment (aboedo) [reply Purchases/Purchasing/Purchases.swift:1289]: great catch! I'll fix it in #1253

**Pull request merged**